src/src/com/microsoft/aad/adal/DateTimeAdapter.java

```diff
@@ -50,12 +50,31 @@
             DateFormat.DEFAULT);
 
     private final DateFormat iso8601Format = buildIso8601Format();
```

Comment:
But the dates that failed on parsing are very recent (t least all that are in telemetry which we added recently are pretty fresh). I'll send you the query for this type of error